### PR TITLE
MDTZ-980: [PR#1] Use `/v1/files/:fileKey` endpoint to fetch File-based and Node-based designs

### PR DESCRIPTION
**Pull request by akostevich-atlassian**
Within MDTZ-980, we are going to use `lastModified` from top-level Figma nodes as `lastUpdated` for ingested designs.

While this this PR does not use `lastModified` from top-level Figma nodes yet, it creates the foundation for introducing and testing these changes. The next PR will deliver the remaining functionality.

## Changes

- **[Feature]:** Use Figma's `/v1/files/:fileKey` endpoint to fetch File-based and Node-based designs. Remove the integration of `/v1/files/:fileKey/nodes` endpoint from the codebase.
  - When fetch File-based design, use `/v1/files/:fileKey` with `depth=1` query parameter to minimize the payload and latency.
  - When fetch Node-based design, use `/v1/files/:fileKey` with `ids` query parameter to query only nodes from the root (where `lastModified` is available) to the target node.
- **[Improvement]:** Move `FigmaClient` to `infrastructure/figma/figma-client` and reorganise the module for consistency with `JiraClient` (which resides in `infrastructure/jira/jira-client`).
- **[Improvement]:** Split `figma-transformer` into `figma-file-transformer` and `figma-node-transformer` to simplify code navigation and integration of the coming changes,.
- **[Improvement]:** Refactor integration tests for `/associateEntity` and `/disassociateEntity` endpoints to remove implicit dependencies among different component setup (caused by accidental usage of the same constants). Use generated random data to prevent implicit dependencies and guarantee the completeness of test setup.
- **[Improvement]:** Add new integration tests for `/associateEntity` and `/disassociateEntity` endpoints to test the flows for both File-based and Node-based design.

### CODE REVIEW

File: src/domain/entities/testing/mocks.ts [hunk 1]
```diff
@@ -31,20 +31,41 @@ export const MOCK_FIGMA_NODE_IDENTITY = new FigmaDesignIdentity(
...
+export const generateFigmaFileName = () => uuidv4();
+
+export const generateFigmaFileKey = () =>
+	Buffer.from(uuidv4()).toString('base64');
+
+export const generateFigmaNodeId = () =>
+	`${getRandomInt(1, 100)}:${getRandomInt(1, 100)}`;
+
 export const generateFigmaDesignUrl = ({
 	fileKey = MOCK_FIGMA_FILE_KEY,
```
Comment (jv3nk) [inline src/domain/entities/testing/mocks.ts:49]: I think this should probably be using `generateFigmaFileKey` instead of `MOCK_FIGMA_FILE_KEY`.

Comment (akostevich-atlassian) [reply src/domain/entities/testing/mocks.ts]: Yes, great catch! 

✅ Fixed.

File: src/infrastructure/figma/transformers/figma-file-transformer.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	FigmaDesignIdentifier,
+} from '../../../domain/entities';
+import type { FileResponse } from '../figma-client';
+
+type TransformFileToAtlassianDesignParams = {
+	readonly fileKey: string;
+	readonly fileResponse: FileResponse;
+};
+
+export const transformFileToAtlassianDesign = ({
```
Comment (akostevich-atlassian) [inline src/infrastructure/figma/transformers/figma-file-transformer.ts:21]: **note:** This code has been moved from `infrastructure/figma/figma-transformer.ts` with no changes.

File: src/infrastructure/figma/transformers/figma-node-transformer.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+} from '../../../domain/entities';
+import type { FileResponse, Node, NodeDevStatus } from '../figma-client';
+
+type TransformNodeToAtlassianDesignParams = {
+	readonly fileKey: string;
+	readonly nodeId: string;
+	readonly fileResponseWithNode: FileResponse;
+};
+
+export const transformNodeToAtlassianDesign = ({
```
Comment (akostevich-atlassian) [inline src/infrastructure/figma/transformers/figma-node-transformer.ts:22]: **note:** This code has been moved from `infrastructure/figma/figma-transformer.ts` and updated to work with `/v1/files/:fileKey?ids=:id` response.

File: src/infrastructure/figma/figma-client/types.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
+export type GetOAuth2TokenResponse = {
```
Comment (akostevich-atlassian) [inline src/infrastructure/figma/figma-client/types.ts:1]: note: This code has been moved from `infrastructure/figma-client.ts` with almost no changes.

**Review by jv3nk — APPROVED** [review]
This looks great, really nice job fixing up our mock factories. Should be a much lower chance of seeing subtle bugs due to implicit dependencies.

A lot of the stuff added here would be useful in https://github.com/atlassian-labs/figma-for-jira/pull/65, so I think it'd be good to get this merged in first.

**Pull request merged**